src/components/PopupWithForm.js

```diff
@@ -23,10 +36,28 @@ export class PopupWithForm extends Popup {
     super.setEventListeners();
 
     this.form.addEventListener("submit", (evt) => {
+      //
+      renderLoading(true, this.form);
+
       evt.preventDefault();
 
       this.handleFormSubmit(this._getInputValues());
       this.close();
     });
   }
+
+  //
+  setDeleteListener() {
+    super.setEventListeners();
+    this.form.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this.handleFormSubmit(this.cardElement, this.id);
+      this.close();
+    });
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`